packages/dds/tree/src/rebase/rebaser.ts

```diff
@@ -167,14 +167,26 @@ export interface ChangeRebaser<TChangeset> {
      * - `rebase(a, compose([]))` is equal to `a`.
      * - `rebase(compose([]), a)` is equal to `a`.
      */
-    rebase(change: TChangeset, over: TChangeset): TChangeset;
+    rebase(change: TChangeset, over: ChangeWithMetadata<TChangeset>): TChangeset;
 
     // TODO: we are forcing a single AnchorSet implementation, but also making ChangeRebaser deal depend on/use it.
     // This isn't ideal, but it might be fine?
     // Performance and implications for custom Anchor types (ex: Place anchors) aren't clear.
     rebaseAnchors(anchors: AnchorSet, over: TChangeset): void;
+
+    filterReferences(change: TChangeset, shouldRemoveReference: (revision: RevisionTag) => boolean): TChangeset;
+}
+
+export interface ChangeWithMetadata<TChangeset> {
```

Comment:
Are you expecting we may want to add more metadata in the future (other than the revision I mean)?
If not, we could rename this `TaggedChange` for brevity. Not a big deal either way.

Reply:
It's possible we'll want to include more information in the future, but TaggedChange might be a nicer name anyway.